### PR TITLE
arm-performance-libraries 24.04 (new cask)

### PR DESCRIPTION
**Pull request by pthariensflame**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=pullrequests).
- [x] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [x] `brew audit --cask --new <cask>` worked successfully.
- [x] `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --cask <cask>` worked successfully.
- [x] `brew uninstall --cask <cask>` worked successfully.

### CODE REVIEW

File: Casks/a/arm-performance-libraries.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
+cask "arm-performance-libraries" do
+  version "24.04,17145f380ea64fe6b26de596182561bc,17145f38-0ea6-4fe6-b26d-e596182561bc"
+  install_suffix="#{version.csv.first}_flang-new_clang_18"
+  sha256 "228bf3a2c25dbd45c2f89c78f455ee3c7dfb25e121c20d2765138b5174e688dc"
+
+  url "https://developer.arm.com/-/media/Files/downloads/hpc/arm-performance-libraries/#{version.csv.first.major}-#{version.csv.first.minor}/macos/arm-performance-libraries_#{version.csv.first}_macOS.tgz?rev=#{version.csv.second}&revision=#{version.csv.third}"
```
Comment (razvanazamfirei) [inline Casks/a/arm-performance-libraries.rb:6]:
```suggestion
  url "https://developer.arm.com/-/media/Files/downloads/hpc/arm-performance-libraries/#{version.dots_to_hyphens}/macos/arm-performance-libraries_#{version}_macOS.tgz"
```

File: Casks/a/arm-performance-libraries.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
+cask "arm-performance-libraries" do
+  version "24.04,17145f380ea64fe6b26de596182561bc,17145f38-0ea6-4fe6-b26d-e596182561bc"
```
Comment (razvanazamfirei) [inline Casks/a/arm-performance-libraries.rb:2]:
```suggestion
  version "24.04"
```

File: Casks/a/arm-performance-libraries.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  install_suffix="#{version.csv.first}_flang-new_clang_18"
```
Comment (razvanazamfirei) [inline Casks/a/arm-performance-libraries.rb:3]:
```suggestion
  install_suffix="#{version}_flang-new_clang_18"
```

**Review by razvanazamfirei — CHANGES_REQUESTED** [review]
Hi @pthariensflame, is there any reason why the simplified version scheme would not work?

**Comment by pthariensflame** [comment]
> Hi @pthariensflame, is there any reason why the simplified version scheme would not work?

No specific reason, no!  It just makes autobump from livecheck harder, but I guess that was out of the question anyways given the install suffix issue.  I'll make that change when I can next touch my dev machine!

**Comment by pthariensflame** [comment]
@razvanazamfirei As it turns out, the revision info is not needed for the download URL, so I got rid of it entirely!  Thank you for making me look at it more closely!

**Comment by pthariensflame** [comment]
@razvanazamfirei Any news?

File: Casks/a/arm-performance-libraries.rb [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+  uninstall delete: [
+              "/opt/arm/arm-performance-libraries_#{install_suffix}",
+              "/opt/arm/armpl_#{install_suffix}",
+            ],
+            rmdir:  "/opt/arm"
+
+  # No zap stanza required
+
+  caveats "You may need to source /opt/arm/armpl_#{install_suffix}/armpl_env_vars.sh in order to use these libraries."
```
[the remark this answers is not in the document]
Comment (pthariensflame) [reply Casks/a/arm-performance-libraries.rb:34]: Applies across-the-board.  The script just sets some environment variables, so in principle a user could do that just as well manually, but running it is vender-recommended practice on all installations.

File: Casks/a/arm-performance-libraries.rb [hunk 5]
```diff
@@ -0,0 +1,35 @@
+cask "arm-performance-libraries" do
+  version "24.04"
+  install_suffix="#{version}_flang-new_clang_18"
+  sha256 "228bf3a2c25dbd45c2f89c78f455ee3c7dfb25e121c20d2765138b5174e688dc"
+
+  url "https://developer.arm.com/-/media/Files/downloads/hpc/arm-performance-libraries/#{version.dots_to_hyphens}/macos/arm-performance-libraries_#{version}_macOS.tgz"
+  name "arm-performance-libraries"
```
Comment (bevanjkay) [inline Casks/a/arm-performance-libraries.rb:7]:
```suggestion
  name "Arm Performance Libraries"
```

File: Casks/a/arm-performance-libraries.rb [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    regex(
+      /href=.*?arm-performance-libraries_v?(\d+(?:\.\d+)+)_macOS\.t/i,
+    )
+  end
+
+  depends_on arch: :arm64
+
+  installer script: {
+    executable: "armpl_#{install_suffix}_install.sh",
+    args:       ["-y"],
```
Comment (bevanjkay) [inline Casks/a/arm-performance-libraries.rb:22]: I'm on the fence on whether we should explicitly accept the license systematically. Waiting for some other maintainers to chime in their thoughts.

Comment (razvanazamfirei) [reply Casks/a/arm-performance-libraries.rb:22]: We should be able to use the license caveat, I think. See my last commit.

**Review by razvanazamfirei — APPROVED** [review]

**Review by krehel — APPROVED** [review]

**Review by chenrui333 — APPROVED** [review]

**Pull request merged**